### PR TITLE
[NG] DnD final PR

### PR DESCRIPTION
**Pull request by Shijir**
- [fix selection error during dragging in IE11](https://github.com/vmware/clarity/pull/2524/files#diff-e6ed458ed9d31a9a67890113e007f404R67)
- [fix error message (just removed backticks in the messages)](https://github.com/vmware/clarity/pull/2524/files#diff-267db36747c7d2123571879a446db50fR23)
- created droppable demos

Tested on: Chrome, Firefox, Safari, Edge, IE

Signed-off-by: Shijir Tsogoo <stsogoo@vmware.com>

### CODE REVIEW

File: src/app/drag-and-drop/basic-droppable.demo.html [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    <div class="radio-inline">
+        <input type="radio" id="drag-handle" name="droppable-variant">
+        <label for="drag-handle">Drag Handle</label>
+    </div>
+
+</div>
+
+<div class="clr-example" *ngIf="activeDemoVariant==='basic-draggable'">
+
+    <div class="row">
```
Comment (gnomeontherun) [inline src/app/drag-and-drop/basic-droppable.demo.html]: I guess this is the old branch, but you'll need to use the new grid when you rebase.

File: src/app/drag-and-drop/drag-and-drop.demo.scss [hunk 2]
```diff
@@ -2,4 +2,28 @@
 // This software is released under MIT license.
 // The full license information can be found in LICENSE in the root directory of this project.
 
-@import "../_utils/code";
\ No newline at end of file
+@import "../_utils/code";
```
Comment (gnomeontherun) [inline src/app/drag-and-drop/drag-and-drop.demo.scss]: Don't need this, its removed later anyways.

File: src/app/drag-and-drop/grouping.demo.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import {Component} from "@angular/core";
+import {ClrDragEvent} from "../../clr-angular/utils/drag-and-drop/drag-event";
+
+const FILES = [
+    {name: "img_001.jpg", group: "image-file"}, {name: "mov_001.mov", group: "movie-file"},
+    {name: "aud_001.mp4", group: "audio-file"}
+];
+
+@Component(
+    {selector: "basic-droppable-demo", styleUrls: ["./drag-and-drop.demo.scss"], templateUrl: "./grouping.demo.html"})
```
Comment (gnomeontherun) [inline src/app/drag-and-drop/grouping.demo.ts]: formatting?

File: src/clr-angular/utils/drag-and-drop/providers/drag-event-listener.ts [hunk 4]
```diff
@@ -61,8 +61,14 @@ export class ClrDragEventListener<T> {
     private customDragEvent(element: Node, startOnEvent: string, moveOnEvent: string, endOnEvent: string): () => void {
         let moveListener: () => void;
         let endListener: () => void;
+        let selecStartListener: () => void;
 
         return this.renderer.listen(element, startOnEvent, () => {
+            // This is needed to disable selection during dragging in IE11.
+            selecStartListener = this.renderer.listen("document", "selectstart", (selectEvent: Event) => {
+                selectEvent.preventDefault();
```
Comment (gnomeontherun) [inline src/clr-angular/utils/drag-and-drop/providers/drag-event-listener.ts]: Do we want to also stopPropagation() here?

**Comment by gnomeontherun** [comment]
This is shaping up nicely, just a few questions.

**Comment by pumano** [comment]
@Shijir amazing work! I would love to use DnD!

File: src/app/drag-and-drop/basic-droppable.demo.html (hunk 1 above)
Comment (Shijir) [reply src/app/drag-and-drop/basic-droppable.demo.html]: Yes, I am expecting substantial changes during rebasing. I will keep your suggestion in mind. 👌

File: src/clr-angular/utils/drag-and-drop/providers/drag-event-listener.ts (hunk 4 above)
Comment (Shijir) [reply src/clr-angular/utils/drag-and-drop/providers/drag-event-listener.ts]: Yes. A select event shouldn't be triggered at all so I think it's appropriate to disable its propagation too. Also, just realized there is misspelling in "selecStartListener"

File: src/app/drag-and-drop/grouping.demo.ts (hunk 3 above)
Comment (Shijir) [reply src/app/drag-and-drop/grouping.demo.ts]: This is what our old clang format is doing for some reason.

**Pull request merged**